# Phylogenomic timetree-calibrated speciation clocks for _Caenorhabditis_ nematodes reveal slow but disproportionate accumulation of post-zygotic reproductive isolation
Supplementary data files of _Caenorhabditis_ divergence time and speciation clock calculations from Fusca et al. (2025). "_C._ sp. 24" refers to _C. agridulce_ throughout.
## Description of files
### 205 Single Copy Ortholog Alignments
Directory of DNA coding sequence alignments for the 205 groups of orthologous genes that are single-copy in all 51 species used for our primary divergence time estimates (i.e. not including _C. auriculariae_ or _C. niphades_). Each file contains one sequence for each of the 51 species. These 205 FASTA files are subdivided into four non-overlapping sets, corresponding to the four gene subsets shown in Figure 2D. Species are referred to using 5-letter abbreviations (e.g. "CELEG" for _C. elegans_).
### 692 Single Copy Orthologs Elegans Supergroup
Directory of DNA coding sequence alignments for the 692 groups of orthologous genes that are single-copy in the 31 species belonging to the Elegans supergroup clade, out of the 51 species that were used for our primary divergence time estimates (i.e. not including _C. niphades_). Each file contains one sequence for each of these 31 species. These 692 FASTA files are subdivided into four non-overlapping sets, corresponding to the four gene subsets shown in Figure 2E. Species are referred to using 5-letter abbreviations (e.g. "CELEG" for _C. elegans_).
### Dated Species Trees
Directory of dated phylogeny files, in NEXUS format. <b>All branch lengths are in units of generations x 10<sup>7</sup> (i.e. tens of millions of generations).</b> In addition to the tree topology and all branch lengths, these phylogeny files also contain branch labels and node labels produced by BEAST (e.g. 95% HPD intervals for node ages and branch relative substitution rates).

<ul>
<li><b>205_orthologs_gamma_priors_relaxed_clock_species_tree.tree</b>: The primary set of divergence time estimates (i.e. those shown in Figure 1), using the 205 single-copy genes and our primary ASTRAL tree topology. </li>
<li><b>partition_1_51_orthologs_gamma_priors_relaxed_clock_species_tree.tree</b>: Divergence time estimates produced using the 51 genes in Set 1 of the 205 single-copy genes (red dots in Figure 2D) and our primary ASTRAL tree topology. </li>
<li><b>partition_2_51_orthologs_gamma_priors_relaxed_clock_species_tree.tree</b>: Divergence time estimates produced using the 51 genes in Set 2 of the 205 single-copy genes (blue dots in Figure 2D) and our primary ASTRAL tree topology. </li>
<li><b>partition_3_51_orthologs_gamma_priors_relaxed_clock_species_tree.tree</b>: Divergence time estimates produced using the 51 genes in Set 3 of the 205 single-copy genes (green dots in Figure 2D) and our primary ASTRAL tree topology. </li>
<li><b>partition_4_52_orthologs_gamma_priors_relaxed_clock_species_tree.tree</b>: Divergence time estimates produced using the 52 genes in Set 4 of the 205 single-copy genes (purple dots in Figure 2D) and our primary ASTRAL tree topology. </li>
<li><b>Elegans_supergroup_old_205_orthologs_gamma_priors_relaxed_clock_species_tree.tree</b>: Divergence time estimates produced using the Elegans supergroup sequences of the 205 genes that were used in our primary date estimates (red dots in Figure 2E) and the topology of the Elegans supergroup from our primary ASTRAL tree topology. </li>
<li><b>Elegans_supergroup_set_1_gamma_priors_relaxed_clock_species_tree.tree</b>: Divergence time estimates produced using the Elegans supergroup sequences of the 163 genes in Set 1 of the 692 Elegans supergroup single-copy genes (blue dots in Figure 2E) and the topology of the Elegans supergroup from our primary ASTRAL tree topology. </li>
<li><b>Elegans_supergroup_set_2_gamma_priors_relaxed_clock_species_tree.tree</b>: Divergence time estimates produced using the Elegans supergroup sequences of the 162 genes in Set 2 of the 692 Elegans supergroup single-copy genes (green dots in Figure 2E) and the topology of the Elegans supergroup from our primary ASTRAL tree topology. </li>
<li><b>Elegans_supergroup_set_3_gamma_priors_relaxed_clock_species_tree.tree</b>: Divergence time estimates produced using the Elegans supergroup sequences of the 162 genes in Set 3 of the 692 Elegans supergroup single-copy genes (purple dots in Figure 2E) and the topology of the Elegans supergroup from our primary ASTRAL tree topology. </li>
<li><b>alternate_topology_1_partition_3_51_orthologs_gamma_priors_relaxed_clock_species_tree.tree</b>: Divergence time estimates produced using the 51 genes in Set 3 of the 205 single-copy genes and alternate tree topology #1 (red dots in Figure 2F). </li>
<li><b>alternate_topology_2_partition_3_51_orthologs_gamma_priors_relaxed_clock_species_tree.tree</b>: Divergence time estimates produced using the 51 genes in Set 3 of the 205 single-copy genes and alternate tree topology #2 (blue dots in Figure 2F). </li>
<li><b>alternate_topology_3_partition_3_51_orthologs_gamma_priors_relaxed_clock_species_tree.tree</b>: Divergence time estimates produced using the 51 genes in Set 3 of the 205 single-copy genes and alternate tree topology #3 (green dots in Figure 2F). </li>
<li><b>alternate_topology_4_partition_3_51_orthologs_gamma_priors_relaxed_clock_species_tree.tree</b>: Divergence time estimates produced using the 51 genes in Set 3 of the 205 single-copy genes and alternate tree topology #4 (purple dots in Figure 2F). </li>
</ul>

### BEAST XML Input Files
Directory of XML files used as input for runs of BEAST. Each XML input file corresponds to the BEAST run for one of the sets of divergence times listed above.
